hepcrawl/spiders/elsevier_spider.py

```diff
@@ -0,0 +1,1079 @@
+                    ".//sb:edited-book/sb:title/ce:inter-ref/text()"
+                ).extract_first()
+        else:
+            book_title = ref.xpath(
+                ".//sb:book//sb:maintitle/text()").extract_first()
+        return book_title
+
+    @staticmethod
+    def _fix_node_text(text_nodes):
+        """Text might be split to multiple elements, this joins them.
+
+        Input must me a list. Returns a string."""
+        # NOTE: should this be more general?
+        title = " ".join(" ".join(text_nodes).split())
+        return title
+
+    @staticmethod
+    def _get_ref_volume(ref):
+        """Get the reference volume. Take only numbers."""
+        volumes_raw = ref.xpath(".//sb:volume-nr/text()").extract()
+        volumes = []
+        for vol in volumes_raw:
+            if "vols" in vol.lower():
+                # NOTE: scrapy doesn't handle (&ndash;) in the file properly?
+                vol_nums = [v for v in vol.split() if has_numbers(v)]
+                if vol_nums:
+                    for vol_num in vol_nums:
+                        volumes.append(vol_num)
+            else:
+                vol_num = get_first([v for v in vol.split() if has_numbers(v)])
+                volumes.append(vol_num)
+
+        return ", ".join(volumes)
+
+    @staticmethod
+    def _get_ref_years(ref):
+        """Get the reference year.
+
+        Return a formatted string if multiple volumes with multiple years.
+        """
+        host = ref.xpath(".//sb:host")
+        years = host.xpath(".//sb:date/text()").extract()
+        if host and years and len(host) > 1:
+            # If reference is contained in multiple hosts, e.g. reprinted.
+            return ", ".join(years)
+        elif host and years:
+            if len(years) == 1:
+                year = years[0]
+            elif len(years) == 2:
+                year = years[0] + "-" + years[1]
+            else:
+                year = ", ".join(years)
+            return year
+
+    def _parse_references(self, ref, label):
+        """Parse all the references."""
+        references = {}
+        textref = ref.xpath(".//ce:textref//text()").extract()
+        sublabel = ref.xpath(".//ce:label//text()").extract_first()
+        if label:
+            if sublabel:
+                sublabel = sublabel.strip("[]")
+                if sublabel != label:
+                    label = label + sublabel
+            references["label"] = label
+        if textref:
+            references["raw_reference"] = self._fix_node_text(textref)
+            return references
+        doi = ref.xpath(".//ce:doi/text()").extract_first()
+        fpage = ref.xpath(".//sb:first-page/text()").extract_first()
+        lpage = ref.xpath(".//sb:last-page/text()").extract_first()
+        publication = self._get_ref_journal_title(ref)
+        title = self._get_ref_title(ref)
+        book_title = self._get_ref_book_title(ref, title)
+        # NOTE: do we need the book edition:
```

Comment:
@bittirousku @annetteholtkamp Does the book edition fit in the new reference structure?